### PR TITLE
Never include Closure Compiler polyfills in codebase.

### PR DESCRIPTION
**Pull request by cramforce**
- they are pretty complex (relying on a symbol polyfill).
- we actually had specific polyfills we need included ourselves duplicating the bytes.

Aspect of #5792

### CODE REVIEW

**Comment by cramforce** [comment]
I'll refactor the way we include polyfills a bit.

File: build-system/tasks/compile.js [hunk 1]
```diff
@@ -227,8 +220,7 @@ function compile(entryModuleFilenames, outputDir,
         // Transpile from ES6 to ES5.
         language_in: 'ECMASCRIPT6',
         language_out: 'ECMASCRIPT5',
-        rewrite_polyfills: !!(
-            options.includePolyfills || options.includeBasicPolyfills),
+        rewrite_polyfills: false,
```
Comment (erwinmombay) [inline build-system/tasks/compile.js]: is `true` the default? (can't remember)

**Comment by cramforce** [comment]
Probably good to document it clearly. I'm about to push a change with some
written docs as well.

File: build-system/dep-check-config.js [hunk 2]
```diff
@@ -130,8 +133,29 @@ exports.rules = [
...
   {
-    filesMatching: 'extensions/**/*.js',
+    filesMatching: '**/*.js',
     mustNotDependOn: 'src/polyfills/**/*.js',
+    whitelist: [
+      // DO NOT add extensions/ files
+      '3p/polyfills.js->src/polyfills/math-sign.js',
+      '3p/polyfills.js->src/polyfills/object-assign.js',
+      'src/polyfills.js->src/polyfills/*.js',
+      'src/polyfills.js->src/polyfills/document-contains.js',
```
Comment (jridgewell) [inline build-system/dep-check-config.js]: Do we need to list each one individually?

Comment (cramforce) [inline build-system/dep-check-config.js]: Yeah, I think it is a feature.
We support wildcards on the left side of the `->` but not the right side.

Comment (jridgewell) [inline build-system/dep-check-config.js]: Remove the line above then?

**Comment by cramforce** [comment]
Arghh, merge conflict in my editor :)

On Thu, Oct 27, 2016 at 11:28 AM, Justin Ridgewell <notifications@github.com

**Pull request merged**